libr/anal/cc.c

```diff
@@ -211,6 +211,30 @@ R_API void r_anal_cc_set_error(RAnal *anal, const char *convention, const char *
 	r_strbuf_fini (&sb);
 }
 
+#if R2_580
+R_API int r_anal_cc_max_arg(RAnal *anal, const char *cc, void *oldDB, char *oldCC, int *oldArg) {
+	int i = 0;
+	r_return_val_if_fail (anal && DB && cc, 0);
+	*oldDB = NULL;
+	*oldCC = NULL;
+	*oldArg = 0;
+	if (oldDB == DB && !strcmp (cc, oldCC)) {
+		return oldArg;
+	}
+	oldDB = DB;
+	free (oldCC);
+	oldCC = strdup (cc);
+	for (i = 0; i < R_ANAL_CC_MAXARG; i++) {
+		r_strf_var (query, 64, "cc.%s.arg%d", cc, i);
+		const char *res = sdb_const_get (DB, query, 0);
+		if (!res) {
+			break;
+		}
+	}
+	oldArg = i;
+	return i;
+}
+#else
 R_API int r_anal_cc_max_arg(RAnal *anal, const char *cc) {
 	int i = 0;
 	r_return_val_if_fail (anal && DB && cc, 0);
```

Comment:
remove the R_TH_LOCALs below. they can be cached inside the RAnal->ccmax_cache struct